format_spec.md

```diff
@@ -0,0 +1,339 @@
+- `scm` - The source code of a Project, for example a GitHub URL.
+- `issues` - An issue tracker where bugs and feature requests can be filed.
+- `community` - A forum or other online discussion platform where users can discuss the Project.
+
+Other values MAY be used, such as `bitbucket` or `github`. However, it is RECOMMENDED to use generic terminology instead of specific names. If you are unsure whether a value COULD become standardized, please prefix it with `x-`.
+
+##### `name`
+
+A human-readable name of the link. For example, "Source Code" or "Project Website".
+
+##### `uri`
+
+The link, to be viewed in a web browser.
+
+---
+
+### `successor` and `predecessor` (optional)
+
+If an older or newer version of a Project exists, but has a separate ID, these fields may be used. These fields MUST be either an array or String representing the respective projects' IDs. If the field is an array, the Projects should be listed in chronological order from when they were first released.
+
+```
+"successor": [
+  "first-successor",
+  "second-successor
+]
+```
+
+In the example above, the project `first-successor` was released first. After the release of `first-successor`, the project `second-successor` was released.
+
+---
+
+## Version Information
+
+Projects can contain a `versions` array, which lists Version Objects.
+
+## Version Objects
+
+Version Objects contain fields relating to a certain version of a Project. Below are all the fields that a Version Object may contain.
+
+### `id`
+
+The ID of this version. There is no required format for how to format a version ID, other than that they CANNOT include spaces.
+
+---
+
+### `name`
+
+The human-readable name of this version. It is RECOMMENDED to follow Semantic Versioning, but it is not required. Spaces MAY be used in this name.
+
+---
+
+### `semver` (optional)
+
+An optional Semantic Versioning compatible version number. This field SHOULD ONLY be present if the version's name conforms to Semantic Versioning. This field SHOULD NOT be present if the name does not conform to SemVer.
+
+---
+
+### `releaseDate`
+
+The release date of this version. It MUST be stored as an ISO-8601 conforming String.
+
+---
+
+### `changelog`
+
+The changelog of this version. This MAY be moved into a separate API call for hosts. The changelog MAY contain HTML.
+
+---
+
+### `minecraftVersions`
+
+This field MUST be an array listing compatible Minecraft versions for this Project version.
+
+---
+
+### `dependencies` (optional)
+
+This field MUST be an array containing information about dependencies that may or may not be required for this project. Each object in this array contains the following fields:
+
+##### `type`
+
+The type of dependency this is. Valid options are:
+
+- `asset` - For example, a required library mod
+- `framework` - For example, a required modloader
+
+Frameworks exist as a separate type because they may require a more advanced installation process.
```

Comment:
Isn't that a problem for the framework's project entry to deal with, not the project using the framework?

Reply:
Frameworks are separate because they may not have a project entry. I personally don't think Forge will provide MCIP format-compatible data any time soon.

Reply:
So MCIP isn't going to index it?

Reply:
I thought about this more and it may be better to store them as projects. I'll add some more fields and values to support it.